velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1399,4 +1400,15 @@ struct TimeZoneMinuteFunction : public TimestampWithTimezoneSupport<T> {
   }
 };
 
+template <typename T>
+struct ToISO8601Function {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Date>& date) {
+    result = DATE()->toISO8601String(date);
```

Comment:
DATE() returns a copy of a shared_ptr which involves atomic increment / decrement of a reference count. It is better to add a static method on DateType class.

```
DateType::toIso8601(date);
```

The 'String' suffix in toISO8601String is redundant since IS0 8601 defines a string format. Also, ISO needs to be Iso to match coding guidelines. Consider renaming to toIso8601.